lib/jsonapi/request_parser.rb

```diff
@@ -265,10 +265,7 @@ def parse_filters(filters)
             return @errors.concat(Exceptions::FilterNotAllowed.new(filter_method).errors)
           end
 
-          # Supports only just-included or top-level of multiple-relationship includes, e.g.,
-          # ?include=parent,parent.child&filter[parent.arms]=3
-          filtered_relationship_included = @include_directives.model_includes.include?(relationship.name.to_sym) || @include_directives.model_includes.any? { |include| include.key?(relationship.name.to_sym) }
-          unless filtered_relationship_included
+          unless @include_directives.include_config(relationship.name.to_sym).present?
```

Comment:
This more cleanly handles both of the above cases, as well as when the relationship is not included in a `model_includes`, as is the case when `eager_load_on_include` is false.